driver/compile_common.mli

```diff
@@ -42,9 +42,13 @@ val with_info :
    calling [with_info] several times.
 *)
 
+module Parse_result : sig
+  type 'a t = { ast : 'a; info : info }
+end
+
 (** {2 Interfaces} *)
 
-val parse_intf : info -> Parsetree.signature
+val parse_intf : info -> Parsetree.signature Parse_result.t
 (** [parse_intf info] parses an interface (usually an [.mli] file). *)
 
 val typecheck_intf :
```

Comment:
I would propose to the connect the type of  `parse_intf` and `typecheck_intf` by making `typecheck_intf` takes a `Parsetree.signature Parse_result.t` as argument.